snowfall/tools/ali.py

```diff
@@ -196,3 +203,107 @@ def visualize(input: str,
     if ret != 0:
         raise Exception(f'Failed to run\n{cmd}\n'
                         f'The praat script content is:\n{command}')
+
+
+def compute_ali(lang_dir: str,
+                posts: str,
+                output_dir: str,
+                device: torch.device,
+                max_duration: int = 200,
+                output_beam_size: float = 8.0):
+    # First, load lang
+    lang_dir = Path(lang_dir)
+
+    symbol_table = k2.SymbolTable.from_file(lang_dir / 'words.txt')
+    phone_symbol_table = k2.SymbolTable.from_file(lang_dir / 'phones.txt')
+    phone_ids = get_phone_symbols(phone_symbol_table)
+
+    phone_ids_with_blank = [0] + phone_ids
+    ctc_topo = k2.arc_sort(build_ctc_topo(phone_ids_with_blank))
+
+    if not (lang_dir / 'HLG.pt').exists():
```

Comment:
I think this could be refactored to a function and re-used across this script and decode scripts (and possibly others)

```python
def load_or_compile_HLG(lang_dir: Path) -> k2.Fsa: ...
```

Reply:
Thanks! Will refactor it and add options to enable/disable LM rescoring.